### PR TITLE
fix: Handle empty input values from the share project popover

### LINKED ISSUE #1966: Share link name saves even when the link name is empty
Steps:

- Click on share
- It generates a link with a `custom` name.
- Try to edit the name, and keep the name as ampty
- On `Escape` the model closes and sets the name to empty

More info can be found in the screen-recording below.

https://github.com/webstudio-is/webstudio-builder/assets/11075561/38bdc29b-2878-45e5-a0f1-06289e8fdfd3

### PR DESCRIPTION
**Pull request by JayaKrishnaNamburu**
fixes #1966 

The steps to reproduce are in the linked issue. The popover shouldn't close when the name in the input field is empty.

@taylornowotny i would like you to check this one for the error state please, the Input field didn't have a red outline for error as defined in the figma file. I added it now, So, in order to check the effects of this in other place. It would be great if you can check it once.

Screenshot 2023-07-25 at 17 35 51

### CODE REVIEW

**Review by taylornowotny — COMMENTED** [review]
Looks and works great!

File: apps/builder/app/shared/share-project/share-project.tsx [hunk 1]
```diff
@@ -88,15 +98,14 @@ const Menu = ({
...
-            onChange={(event) => {
-              onChangeName(event.target.value);
-            }}
+            color={customLinkName.length === 0 ? "error" : undefined}
+            value={customLinkName}
+            onChange={(event) => setCustomLinkName(event.target.value)}
             onKeyDown={(event) => {
               if (event.key === "Enter") {
-                setIsOpen(false);
+                saveCustomLinkName();
```
Comment (TrySound) [inline apps/builder/app/shared/share-project/share-project.tsx:108]: I would call this on blur too

**Comment by kof** [comment]
Anything we can do to make this unstuck? its been created 2 weeks ago

File: packages/design-system/src/components/input-field.tsx [hunk 2]
```diff
@@ -71,6 +71,7 @@ const containerStyle = css({
 
   "&:has([data-input-field-input][data-color=error])": {
     borderColor: theme.colors.borderDestructiveMain,
+    outline: `solid 2px ${theme.colors.borderDestructiveMain}`,
```
[the remark this answers is not in the document]
Comment (JayaKrishnaNamburu) [reply packages/design-system/src/components/input-field.tsx:74]: Yes, should the input be shown all the time if the value is error or just on the focus ?

**Comment by JayaKrishnaNamburu** [comment]
> Anything we can do to make this unstuck? its been created 2 weeks ago

Nothing much, was just hung up on the cli after this

File: packages/design-system/src/components/input-field.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (TrySound) [reply packages/design-system/src/components/input-field.tsx:74]: I mean border is enough when input is not focused. The problem here is that outline overlaps red border and user no longer see the error state. So here you can make outline red as well to not loose the information and keep input style consistent.

**Comment by kof** [comment]
@JayaKrishnaNamburu any reason this got stuck?

**Comment by kof** [comment]
@JayaKrishnaNamburu lets finish this one

**Comment by JayaKrishnaNamburu** [comment]
Done, @TrySound removed `outline` since `border` is enough. And now saving the url on blur too. Let me know, if there are any changes needed.

File: packages/design-system/src/__generated__/figma-design-tokens.json [hunk 3]
```diff
@@ -1451,8 +1451,6 @@
   },
   "$themes": [],
   "$metadata": {
-    "tokenSetOrder": [
-      "global"
-    ]
+    "tokenSetOrder": ["global"]
   }
-}
```
Comment (JayaKrishnaNamburu) [inline packages/design-system/src/__generated__/figma-design-tokens.json:1458]: This change came from the pre-commit tasks

File: apps/builder/app/shared/share-project/share-project.tsx [hunk 4]
```diff
@@ -107,16 +117,16 @@ const Menu = ({
           <Item>
             <Label>Name</Label>
             <InputField
-              value={name}
-              onChange={(event) => {
-                onChangeName(event.target.value);
-              }}
+              color={customLinkName.length === 0 ? "error" : undefined}
+              value={customLinkName}
+              onChange={(event) => setCustomLinkName(event.target.value)}
```
Comment (kof) [inline apps/builder/app/shared/share-project/share-project.tsx:122]: I would trim it as well
```suggestion
              onChange={(event) => setCustomLinkName(event.target.value.trim())}
```

**Review by kof — APPROVED** [review]

**Pull request merged**